content/concepts/asset-pricing.md

```diff
@@ -42,6 +42,28 @@ The image below shows how to set the Dynamic pricing of an asset in the Ocean's
 
 Ocean Protocol also allows publishers to set the pricing using ocean.js and ocean.py library.
 
+### Asset price
+
+#### Action: Add liquidity from pool
+
+With one-sided staking, when liquidity is added to the pool, the ocean tokens are added to the liquidity pool. To protect funds from impermanent loss due to changes in the ratio of tokens in the liquidity pool, Ocean protocol's bot mints new datatokens and adds them to the pool. Thus, when liquidity is added to the pool, the ratio of tokens remains constant, and there is no price impact on the datatoken.
+
+#### Action: Remove liquidity from pool
+
+When the liquidity is removed from the pool, the Ocean tokens are returned to the respective address. The address only receives Ocean tokens due to one-sided staking. Ocean protocol's bot burns the datatokens from the pool to protect funds from impermanent loss due to changes in the ratio of tokens in the liquidity pool. Thus, in this case, there is also no price impact on the datatoken.
+
+#### Action: Buy datatoken
+
+When a datatoken is bought by paying Ocean tokens to the pool, the ratio of Ocean token and datatoken changes, there would be more Ocean tokens and fewer datatokens in the pool than previously. As there would be fewer datatokens, the pool will increase the amount of Ocean tokens required to buy a datatoken in the following transactions to maintain the constant ratio between tokens. Thus, the price of the datatoken increases whenever datatokens are bought.
+
+#### Action: Buy dataset
+
+Buying a dataset requires that the datatokens are swapped from the pool by paying Ocean tokens. Thus, if users buy datatokens, the price of datatokens will increase. However, if users already have the datatokens, they can simply use them to buy the asset or the service without requiring interaction with the pool. In such a case, the price of the datatoken doesn't change.
```

Comment:
Suggestion: "Buying a dataset involves swapping a datatoken from the liquidity pool by paying Ocean tokens. Thus, if users buy datatokens, the price of datatokens will increase. However, if users already have the datatokens, they can use them to buy the asset or the service without requiring interaction with the pool. In such a case, the price of the datatoken doesn't change."